features/auditing.feature

```diff
@@ -1,25 +1,72 @@
 
-  Scenario: As an assessor I can audit the validation requests are sent when the application is invalid
+  Scenario: I can audit the validation requests are sent when the application is invalid
     Given I create an additional document validation request with "Picture of the dog"
     And the planning application is invalidated
     When I view the planning application audit
     Then there is an audit entry containing "invalidation requests have been emailed: Additional document validation request #2, Additional document validation request #1"
+    And there is an audit entry containing "Application invalidated"
 
-  Scenario: As an assessor I can audit when a validation request is cancelled after an application is made invalid
+  Scenario: I can audit when a validation request is cancelled after an application is made invalid
     Given I create an additional document validation request with "Meme of the dog"
     And the planning application is invalidated
     When I view the application's validations requests
     And I cancel a validation request for a "Meme of the dog" with "Meme of dog is no longer needed"
     When I view the planning application audit
     Then there is an audit entry containing "Cancelled: validation request (new document#1)"
     Then there is an audit entry containing "Reason: Meme of dog is no longer needed"
+
+  Scenario: I can view in the audit log when a planning application has been withdrawn
+    Given I press "Cancel application"
+    And I choose "Withdrawn by applicant"
+    And I fill in "Can you provide more detail?" with "Applicant is moving to Bermuda, because heck this"
+    And I press "Save"
+    Then there is an audit entry containing "Application withdrawn"
+    Then there is an audit entry containing "Applicant is moving to Bermuda, because heck this"
+
+  Scenario: I can view in the audit log when a planning application has been returned
+    Given I press "Cancel application"
+    And I choose "Returned as invalid"
+    And I fill in "Can you provide more detail?" with "Applicant sent selfies instead of floor plans"
+    And I press "Save"
+    Then there is an audit entry containing "Application returned"
+    Then there is an audit entry containing "Applicant sent selfies instead of floor plans"
+
+  Scenario: Updating constraints displays all updates in the audit log
+    Given I press "Constraints"
+    And I press "Update"
+    And I check "National Park"
+    And I check "Broads"
+    And I uncheck "Conservation Area"
+    And I press "Save"
+    Then there is an audit entry containing "Constraint added"
+    And there is an audit entry containing "National Park"
+    And there is an audit entry containing "Constraint added"
+    And there is an audit entry containing "Broads"
+    And there is an audit entry containing "Constraint removed"
+    And there is an audit entry containing "Conservation Area"
```

Comment:
Can we merge these pairs? `Constraint added:\nNational Park`? It's not _great_ but it ties it together, rather than having loose bit of texts checked through all audit entries.

Reply:
I've added an extra step to check two values instead of one, does it look a bit better or does it need reworking?